testdata/TestCheckAgainstSpecData.json

```
@@ -259,113 +259,670 @@
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        },
+        {
+          "hash": 1,
+          "prefix": "qrvM3Q=="
+        }
+      ]
+    },
+    "Spec": {
+      "leaf_spec": {
+        "hash": 1,
+        "prehash_value": 1,
+        "length": 1,
+        "prefix": "AA=="
+      },
+      "inner_spec": {
+        "child_order": [
+          0,
+          1
+        ],
+        "child_size": 33,
+        "min_prefix_length": 4,
+        "max_prefix_length": 12,
+        "hash": 1
+      },
+      "max_depth": 4,
```

Shouldn't actually this be 0 so that it uses the default value of 128?

```suggestion
      "max_depth": 0,
```

---

ah yes! thank you